src/main/java/core/basesyntax/FileWork.java

```diff
@@ -1,7 +1,62 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
 public class FileWork {
+    private static final int STOP_READING = -1;
+    private static final int NEW_LINE = 10;
+    private static final int SPACE_CHAR = 32;
+    private static final int SYMBOL_FROM = 65;
+    private static final int SYMBOL_TO = 122;
+    private static final char SEARCHED_SYMBOL = 'w';
+
     public String[] readFromFile(String fileName) {
-        return null;
+        String wholeText = copyTextFromFile(fileName);
+        return findAllWordsWithW(wholeText);
+    }
+
+    private String copyTextFromFile(String fileName) {
+        StringBuilder wholeText = new StringBuilder();
+        try {
+            FileReader fileReader = new FileReader(fileName);
+            BufferedReader bufferedReader = new BufferedReader(fileReader);
+            int symbol = bufferedReader.read();
+
+            while (symbol != STOP_READING) {
+                if (symbol >= SYMBOL_FROM && symbol <= SYMBOL_TO || symbol == SPACE_CHAR) {
+                    wholeText.append((char) symbol);
+                } else if (symbol == NEW_LINE) {
+                    wholeText.append(" ");
+                }
+                symbol = bufferedReader.read();
+            }
+        } catch (FileNotFoundException e) {
+            throw new RuntimeException("File couldn't be found", e);
+        } catch (IOException e) {
+            throw new RuntimeException("Couldn't read a file", e);
+        }
+        return wholeText.toString();
+    }
+
+    private String[] findAllWordsWithW(String wholeText) {
+        if (wholeText.length() != 0) {
+            String[] wholeTextArray = wholeText.toLowerCase().split(" ");
+            Arrays.sort(wholeTextArray);
```

Comment:
let's sort after we excluded the wrong words

Reply:
Done